File: Classement.py

class Classement():

    def __init__(self, nb_place):
        self.top=["" for i in range(nb_place)]
        self.nb_place=nb_place
        self.out=[]
    
    def add_top(self, name, position):
        if self.top[position-1]=="":
            self.top[position-1]=name
        else:
            self.decaler(position-1)
            self.top[position-1]=name

    def add_out(self,name):
        self.out.append(name)
        
    def decaler(self, position):
        self.out.append(self.top[self.nb_place-1])
        for i in range(self.nb_place-2,position-1,-1):
            self.top[i+1]=self.top[i]
    
    def supp_out(self,name):
        self.out.remove(name)

File: MACLE/src/main/java/com/macle/macle/Classement.java
package com.macle.macle;

import javafx.collections.FXCollections;
import javafx.collections.ObservableList;

public class Classement {
    int nb_place;
    ObservableList<String> top, out;

    Classement(int nb_place){
        this.nb_place = nb_place;
        this.top = FXCollections.observableArrayList();
        this.out = FXCollections.observableArrayList();
        for (int i=0; i<nb_place; i++) {
            this.top.add("");
        }
    }

    void add_top(String name, int pos){
        if (pos < 1 || pos > nb_place) return;
        int index = pos - 1;
        if (!this.top.get(index).isEmpty()) {
            this.decaler(index);
        }
        this.top.set(index, name);
    }

    void add_out(String name){
        this.out.add(name);
    }

    void decaler(int pos){
        // Si la dernière place n'est pas vide, l'élément sort du top
        if (!this.top.getLast().isEmpty()) {
            this.out.add(this.top.getLast());
        }
        // Décalage vers le bas
        for (int i = this.nb_place - 2; i >= pos; i--) {
            this.top.set(i + 1, this.top.get(i));
        }
    }

    void supp_out(String name){
        this.out.remove(name);
    }

}
